### PR TITLE
Fix modifying valueless attrs

### PR DESCRIPTION
**Pull request by courajs**
Previously adding a value to a valueless attr would print without the `=`, for example:
`<Input checked />` -> `<Input checked{{true}} />`

### CODE REVIEW

File: src/parse-result.ts [hunk 1]
```diff
@@ -1017,6 +1017,22 @@ export default class ParseResult {
...
+            } else {
+              equals = '=';
+              quote = quote || '"';
+              dirtyFields.delete('isValueless');
+            }
+          } else if (node.isValueless && dirtyFields.has('value')) {
+            equals = '=';
+          }
+
           if (dirtyFields.has('value')) {
```
Comment (dcyriller) [inline src/parse-result.ts:1036]:
```suggestion
          }

          if (dirtyFields.has('value')) {
            equals = '=';
```

File: src/parse-result.test.ts [hunk 2]
```diff
@@ -270,6 +270,33 @@ describe('ember-template-recast', function () {
         ></div>`);
     });
 
+    test('modifying attribute to be valueless', function () {
```
Comment (dcyriller) [inline src/parse-result.test.ts:273]: These tests are in the `ElementNode` section. May you move them to `AttrNode`?

File: src/parse-result.ts [hunk 3]
```diff
@@ -1017,6 +1017,22 @@ export default class ParseResult {
...
+          if (dirtyFields.has('isValueless')) {
+            if (node.isValueless) {
+              equals = '';
+              quote = '';
+              valueSource = '';
+              dirtyFields.delete('isValueless');
+              dirtyFields.delete('value');
+            } else {
+              equals = '=';
+              quote = quote || '"';
```
Comment (dcyriller) [inline src/parse-result.ts:1029]: We'd like to make `quote` depend also  on the `node.value.type`. If type is a MustacheStatement, we prefer to have a `""` quote.

Comment (dcyriller) [reply src/parse-result.ts:1029]: We could add this test case:
```js
      let template = '<Foo data-foo />';

      let ast = parse(template) as any;
      ast.body[0].attributes[0].isValueless = false;
      ast.body[0].attributes[0].value = builders.mustache('true');

      expect(print(ast)).toEqual('<Foo data-foo={{true}} />');
```

**Comment by dcyriller** [comment]
Good catch - thank you for the follow up!

File: src/parse-result.ts (hunk 3 above)
Comment (courajs) [reply src/parse-result.ts]: Great catch, thank you!

**Review by dcyriller — APPROVED** [review]

**Pull request merged**